pkg/sshclient/client.go

```diff
@@ -0,0 +1,88 @@
+package sshclient
+
+import (
+	"fmt"
+	"os/exec"
+
+	"github.com/dcos/dcos-core-cli/pkg/pluginutil"
+
+	"github.com/dcos/dcos-cli/api"
+	"github.com/sirupsen/logrus"
+)
+
+// Client to execute commands over SSH.
+type Client struct {
+	BinaryPath string
+	Args       []string
+	cmd        *exec.Cmd
+	Logger     *logrus.Logger
+}
+
+var baseArgs = []string{
+	"-A",
+	"-t",
+}
+
+// NewClient creates a new client to start a SSH session.
+func NewClient(options []string, config, user, proxy, host string) (*Client, error) {
+	c := &Client{}
+	c.Logger = pluginutil.Logger()
+	binPath, err := exec.LookPath("ssh")
```

Comment:
If the binary path was configurable yet defaulted to `exec.LookPath("ssh")`, it'd be easier to test the client, for example by having a fake SSH binary which would make assertions on how it is invoked.